core/src/main/java/org/commonjava/maven/ext/core/impl/RelocationManipulator.java

```diff
@@ -203,9 +289,298 @@ private void updateDependencyExclusion( ProjectVersionRef depPvr, ProjectVersion
+        }
+
+        if ( build == null )
+        {
+            return Collections.emptyMap();
+        }
+
+        final Map<String, Plugin> result = build.getPluginsAsMap();
+        if ( result == null )
+        {
+            return Collections.emptyMap();
+        }
+
+        return result;
+    }
+
+    private static Map<String, Plugin> getManagedPluginMap( final ModelBase base )
+    {
+        if ( base instanceof Model )
+        {
+            final Build build = ( (Model) base ).getBuild();
+            if ( build == null )
+            {
+                return Collections.emptyMap();
+            }
+
+            final PluginManagement pm = build.getPluginManagement();
+            if ( pm == null )
+            {
+                return Collections.emptyMap();
+            }
+
+            final Map<String, Plugin> result = pm.getPluginsAsMap();
+            if ( result == null )
+            {
+                return Collections.emptyMap();
+            }
+
+            return result;
+        }
+
+        return Collections.emptyMap();
+    }
+
+    /**
+     * Go through the plugin / plugin-execution configurations and find references to the <code>skip</code> parameter for the given Maven plugin
+     * instance.
+     * @return the configurations
+     */
+    private Map<ConfigurationContainer, String> findConfigurations( final Plugin plugin )
+    {
+        if ( plugin == null )
+        {
+            return Collections.emptyMap();
+        }
+
+        final Map<ConfigurationContainer, String> configs = new LinkedHashMap<>();
+        Object configuration = plugin.getConfiguration();
+        if ( configuration != null )
+        {
+            configs.put( plugin, configuration.toString() );
+        }
+
+        final List<PluginExecution> executions = plugin.getExecutions();
+        if ( executions != null )
+        {
+            for ( PluginExecution execution : executions )
+            {
+                configuration = execution.getConfiguration();
+                if ( configuration != null )
+                {
+                    configs.put( execution, configuration.toString() );
+                }
+            }
+        }
+
+        return configs;
+    }
+
+    private Xpp3Dom getConfigXml( final Node node )
+                    throws ManipulationException
+    {
+        final String config = galleyWrapper.toXML( node.getOwnerDocument(), false )
+                                           .trim();
+
+        try
+        {
+            return Xpp3DomBuilder.build( new StringReader( config ) );
+        }
+        catch ( final XmlPullParserException | IOException e )
+        {
+            throw new ManipulationException( "Failed to re-parse plugin configuration into Xpp3Dom: {}. Config was: {}", e.getMessage(), config, e );
+        }
+    }
+
     @Override
     public int getExecutionIndex()
     {
         return 15;
     }
+
+    private final class PluginReference
+    {
```

Comment:
Question : are we needing to store the Node objects in order to be able to update them transparently ?

I think a couple of sentences of javadoc on this class would be useful :-)

Reply:
It might be possible to update inline.

Reply:
OK, it is agreed to make this change.